packages/tables/backend/sirius-components-tables/src/main/java/org/eclipse/sirius/components/tables/Column.java

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.tables;
+
+import java.text.MessageFormat;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.eclipse.sirius.components.annotations.Immutable;
+
+/**
+ * Column concept of the table representation.
+ *
+ * @author arichard
+ * @author lfasani
+ */
+@Immutable
+public final class Column {
+
+    private UUID id;
+
+    private String targetObjectId;
+
+    private String targetObjectKind;
+
+    private UUID descriptionId;
```

Comment:
Our descriptionId have not been UUID in a long time. It's not a major issue for now but you will end up with an issue here when the support for the view DSL will be introduced.